packages/icon/src/vaadin-iconset-mixin.js

```diff
@@ -123,23 +100,75 @@ export const IconsetMixin = (superClass) =>
         initIconsMap(iconset, name);
         iconset.size = size;
         iconset.name = name;
-
-        // Call this function manually instead of using observer
-        // to make it work without appending element to the DOM.
-        iconset.__nameChanged(name);
       }
     }
 
+    /**
+     * The name of the iconset. Every iconset is required to have its own unique name.
+     * All the SVG icons in the iconset must have IDs conforming to its name.
+     *
+     * See also [`name`](#/elements/vaadin-icon#property-name) property of `vaadin-icon`.
+     *
+     * @return {string}
+     */
+    get name() {
+      return this.__name;
+    }
+
+    /**
+     * @type {string}
+     */
+    set name(name) {
+      const oldName = this.__name;
+      this.__name = name;
+      this.__nameChanged(name, oldName);
+    }
+
+    /**
+     * The size of an individual icon. Note that icons must be square.
+     *
+     * When using `vaadin-icon`, the size of the iconset will take precedence
+     * over the size defined by the user to ensure correct appearance.
+     *
+     * @return {number}
+     */
+    get size() {
+      // Use default property value as a fallback here instead of the constructor
+      // to not override an instance property in the lazy upgrade scenario below.
+      return this.__size !== undefined ? this.__size : 24;
+    }
+
+    /**
+     * @type {number}
+     */
+    set size(size) {
+      this.__size = size;
+    }
+
     /** @protected */
     connectedCallback() {
-      super.connectedCallback();
+      // A user may set a property on an _instance_ of an element
+      // before the custom element is lazily imported and upgraded.
+      // If so, we need to run it through the proper class setter.
+      ['name', 'size'].forEach((prop) => {
```

Comment:
This part is needed for `name` to trigger `__nameChanged` which would then update icon instances, and for `size property to be stored in the internal `__size` variable (and to override fallback default value in the getter).

This inspired by "[make properties lazy](https://web.dev/articles/custom-elements-best-practices#make_properties_lazy)" best practice. LitElement also has logic for [saving instance properties](https://github.com/lit/lit/blob/6a232e92af9372892c7a916dd3d25947be674ee0/packages/reactive-element/src/reactive-element.ts#L1063).